pwndbg/gdblib/disasm/arch.py

```diff
@@ -667,19 +670,15 @@ def _enhance_next(
 
     # This is the default implementation.
     # Subclasses should override this for more accurate behavior/to catch more cases. See x86.py as example
-    def _resolve_target(self, instruction: PwndbgInstruction, emu: Emulator | None, call=False):
+    def _resolve_target(self, instruction: PwndbgInstruction, emu: Emulator | None):
         """
         Architecture-specific hook point for _enhance_next.
 
-        Returns the value of the instruction pointer assuming this instruction executes (and any conditional jumps are taken)
-
-        "call" specifies if we allow this to resolve call instruction targets
+        Returns the program counter target of this instruction.
+        Even in the case of conditional jumps, the potential target should be resolved.
```

Comment:
Old comment was incorrect. The determination of whether the _resolve_target can handle `call` instructions is now resolved by the caller of this function (and all subclasses now reflect this), which makes it much simpler and correct.